### ISSUE TITLE
Bulk Export does not work with Partitions

### ISSUE THREAD

**Issue by gansoper** [hapifhir/hapi-fhir#2931]
Prerequisites:

HAPI FHIR 5.4.1
Partitions enabled

Case 1. 

1. GET http://localhost:15443/server/R4/PARTITION_A/$export => job link is received
2.  GET JOB link => Empty JSON without binary links oblserved
      Expected: JSON with links to binary ndjsons present

Case 2. 

1. GET http://localhost:15443/server/R4/PARTITION_A/$export?_type=Patient  => job link is received 
2.  GET JOB link => JSON with links to binary observed
3. GET binary http://127.0.0.1:15443/server/R4/PARTITION_A/Binary/52 
=> 
{
    "resourceType": "OperationOutcome",
    "issue": [
        {
            "severity": "error",
            "code": "processing",
            "diagnostics": "Resource Binary/52 is not known"
        }
    ]
}

Expected: ndjson for patient returned.

**Comment by tadgh** [comment]
You are absolutely right. Currently, Bulk export writes its binaries to the DEFAULT partition. This is clearly a bug, but if you need to fetch that binary at the moment, you can do it by checking `http://127.0.0.1:1544/server/R4/DEFAULT/Binary/52`.

**Comment by kishan-thoughti** [comment]
I have tried bulk export on multitenant enabled fhir server version 6.2.2 but still this issue is not resolved yet. Does anyone has the idea when it is got resolved?
